hnix-store-remote/src/System/Nix/Store/Remote/Logger.hs

```diff
@@ -34,12 +37,23 @@ processOutput = go decoder
           case ctrl of
             e@(Error _ _) -> return [e]
             Last -> return [Last]
+            Read n -> do
```

Comment:
StoreMonad now embeds Maybe Handle in its StateT so we can provide handle and use it to provide data when other side requests tunneling.